src/PIL/Image.py

```diff
@@ -700,7 +700,10 @@ def __array__(self):
         class ArrayData:
             __array_interface__ = new
 
-        return np.array(ArrayData())
+        arr = np.array(ArrayData())
+        if dtype is not None:
+            arr = arr.astype(dtype)
+        return arr
```

Comment:
```suggestion
        return np.array(ArrayData(), dtype)
```
This seems simpler to me.